Gems/Atom/RPI/Assets/ShaderLib/Atom/RPI/MaterialPipelineCallbacks.azsli

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+// The material pipeline uses these functions to call into material type code.
+// The material type may optionally implement any of these functions for custom behavior.
+// Any functions that the material type does not implement will be automatically stubbed out with default behavior.
+
+
+
+// This struct includes fields for all vertex data that can possibly be supported by any material pipeline.
+// The actual vertex stage inputs and outputs will vary between shaders. If any of these fields isn't needed
+// by a given shader, then the material pipeline will initialize them to 0 and will not use them.
+// This way, the material type implementation of the callback function can read and set any of these values
+// without caring whether they are actually used. If the value is not used then any extra code will
```

Comment:
I'm not sure if DCE will necessarily apply here, since if you write 0 to a field which ultimately isn't consumed by the subsequent stage, won't the compiler still need to emit a `mov v0, 0` instruction at the very least and keep the ISBE slot for that data?

Reply:
Hm, you might be right, I'm not sure. I'll remove that part of the comment.